src/main/java/com/ldtteam/structurize/placement/handlers/entity/EntityHandlers.java

```diff
@@ -0,0 +1,226 @@
+     * in other mods just to override one
+     * @param handler the new handler to add
+     * @param override the class to override if it can be found
+     */
+    public static void add(final IEntityHandler handler, final Class<?> override)
+    {
+        for (int i = 0; i < handlers.size(); i++)
+        {
+            if (override.isInstance(handlers.get(i)))
+            {
+                handlers.set(i, handler);
+                return;
+            }
+        }
+        add(handler);
+    }
+
+    /**
+     * Adds a handler to the start of the handlers list,
+     * effectively overriding existing ones with similar
+     * 'canHandle' functions because this one will evaluate before them
+     * @param handler the new handler to add
+     */
+    public static void add(final IEntityHandler handler)
+    {
+        handlers.add(1, handler);
+    }
+
+    /**
+     * Private constructor to hide implicit one.
+     */
+    private EntityHandlers()
+    {
+        // Intentionally left empty.
+    }
+
+    /**
+     * Handler for all kinds of mob
+     */
+    public static class MobEntityHandler implements IEntityHandler
+    {
+        @Override
+        public boolean canHandle(final Entity entity)
+        {
+            return entity instanceof Mob;
+        }
+
+        @Override
+        public boolean canPlace(final Entity entity, final boolean isCreative, final boolean isFancy)
+        {
+            return isCreative;
+        }
+
+        @Override
+        public List<ItemStack> getRequiredItems(final Entity entity)
+        {
+            final SpawnEggItem egg = ForgeSpawnEggItem.fromEntityType(entity.getType());
+            return egg == null ? List.of() : List.of(new ItemStack(egg));
+        }
+    }
+
+    /**
+     * Handler for text displays.
+     */
+    public static class TextDisplayHandler implements IEntityHandler
+    {
+        @Override
+        public boolean canHandle(final Entity entity)
+        {
+            return entity instanceof Display.TextDisplay;
+        }
+
+        @Override
+        public boolean canPlace(final Entity entity, final boolean isCreative, final boolean isFancy)
+        {
+            return isCreative && !isFancy;
+        }
+
+        @Override
+        public List<ItemStack> getRequiredItems(final Entity entity)
+        {
+            return List.of();
+        }
+    }
+
+    /**
+     * Handler for item frames.
+     */
+    public static class ItemFrameHandler extends HangingEntityHandler
+    {
+        @Override
+        public boolean canHandle(final Entity entity)
+        {
+            return entity instanceof ItemFrame;
+        }
+
+        @Override
+        public List<ItemStack> getRequiredItems(final Entity entity)
+        {
+            final List<ItemStack> request = new ArrayList<>();
+            request.add(((ItemFrame) entity).getItem().copyWithCount(1));
+            request.add(new ItemStack(Items.ITEM_FRAME));

```

Comment:
you are missing Items.GLOW_ITEM_FRAME

Reply:
This also matches prior behaviour.  Both ITEM_FRAME and GLOW_ITEM_FRAME are matched by this handler but it only demands ITEM_FRAME as a resource.  The glowing is free.  This matches how signs work (since they don't have a glowing item version).

Reply:
We have been including glow itemframes as costs in Schematics, we probably should match that indep

Reply:
I assume you meant "haven't".  Because we haven't.

https://github.com/ldtteam/Structurize/blob/a6e5b5e1147c1eb08eedc4316d8441e287cc6cb2/src/api/java/com/ldtteam/structurize/api/util/ItemStackUtils.java#L162-L171

This is the code being replaced by this handler.  As you can see, it matches both kinds of item frame but reports only the basic frame as required resources.  This was deliberate because it's how signs also work.

Reply:
I can’t recall I said that. But then again, I can’t recall the recipe for glowing item frames
If it is just a glowstone and an item frame, personally I think it’s fine

Reply:
It requires a glowsquid ink sac (which are fairly rare).  This is also the requirement to make a sign have glowing text.  Which means glowing stuff would be a L3+ item only, most likely -- and people wanted to use them at L1.

Reply:
Technically deep ocean stuff is L4+.  Though they can also be found in cave pools.  Pretty rare there too.

Reply:
https://github.com/ldtteam/Structurize/blob/version/main/src/main/java/com/ldtteam/structurize/client/gui/WindowScan.java#L564-L567
Just saying this was there in code you've removed :D

Reply:
> Just saying this was there in code you've removed :D

Yes, but that was the icon, not the resources.  If you look at the screenshot above you can see it still keeps the glow item frame icon, because that's already the default pickblock behaviour.